### PR TITLE
Upstream `get_basic_block` from ksummarize repo

### PR DESCRIPTION
**Pull request by ehildenb**
This adds function `KProve.get_claim_basic_block`.

- Method `flattenLabel` is renamed to `_flatten_label`.
- Method `KProve._write_claim_definition` is updated with variable naming scheme, and changed to return both the Path that it wrote the definition to and the main module name.
- First a test is added in the IMP language of getting a basic block of a proof.
- Method `KProve.get_claim_basic_block` is implemented and exercised by the test.

This PR takes the implementation wholesale from ksummarize repo, with the intention of eventually simplifying it via JSON RPC.

### CODE REVIEW

File: pyk/src/pyk/integration_tests/test_proofs.py [hunk 1]
```diff
@@ -42,3 +42,45 @@ def test_prove_claim_rule_profile(self):
...
+                                     '      $n |-> N $s |-> 0\n' +                                                                                                   \
+                                     '    </state>\n' +                                                                                                              \
+                                     '  </T>\n' +                                                                                                                    \
+                                     '  <generatedCounter>\n' +                                                                                                      \
+                                     '    _DotVar3\n' +                                                                                                              \
+                                     '  </generatedCounter>\n' +                                                                                                     \
+                                     '</generatedTop>'
```
Comment (ehildenb) [inline pyk/src/pyk/integration_tests/test_proofs.py]: @nishantjr is this how you want it formatted?

Comment (nishantjr) [reply pyk/src/pyk/integration_tests/test_proofs.py]: Yeah, but we don't need the `+`s or the line continuations I think.

Comment (nishantjr) [reply pyk/src/pyk/integration_tests/test_proofs.py]: see here: https://github.com/runtimeverification/k/blob/master/pyk/src/pyk/tests/test_kcfg.py#L338

Comment (nishantjr) [reply pyk/src/pyk/integration_tests/test_proofs.py]:
```suggestion
        post_state_pretty_expected = ('<generatedTop>\n'
                                      '  <T>\n'
                                      '    <k>\n'
                                      '      if ( 0 <=Int N ) { { $s = $s + $n ; $n = $n + -1 ; } while ( 0 <= $n ) { $s = $s + $n ; $n = $n + -1 ; } } else { }\n'
                                      '      ~> _DotVar2\n'
                                      '    </k>\n'
                                      '    <state>\n'
                                      '      $n |-> N $s |-> 0\n'
                                      '    </state>\n'
                                      '  </T>\n'
                                      '  <generatedCounter>\n'
                                      '    _DotVar3\n'
                                      '  </generatedCounter>\n'
                                      '</generatedTop>')
```

I'm not sure if the parens are needed.

**Review by nishantjr — APPROVED** [review]
Looks good, besides the formatting.

File: pyk/src/pyk/integration_tests/test_proofs.py (hunk 1 above)
Comment (ehildenb) [reply pyk/src/pyk/integration_tests/test_proofs.py]: The parens are needed, at least I couldn't fine a way without. Looks much nicer.

**Pull request merged**